### PR TITLE
MBST-15486: Add Reviews and Ratings to API

### PR DESCRIPTION
**Pull request by atish-metabroadcast**
note:
* Ratings have sources.

todo:
* Reviews do not have sources (in Owl Reviews inherit source from content).  Need to add same inherit functionality

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/output/writers/RatingsWriter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.atlasapi.output.EntityListWriter;
+import org.atlasapi.output.EntityWriter;
+import org.atlasapi.output.FieldWriter;
+import org.atlasapi.output.OutputContext;
+
+public class RatingsWriter implements EntityListWriter<Rating> {
+
+    private final EntityWriter<Publisher> sourceWriter;
+
+    public RatingsWriter(@Nonnull EntityWriter<Publisher> sourceWriter) {
```
Comment (MartinsIrbe) [inline atlas-api/src/main/java/org/atlasapi/output/writers/RatingsWriter.java]: Do we need `Nonnull` if we are then using `checkNotNull`?

File: atlas-cassandra/src/main/java/org/atlasapi/content/ContentDeserializationVisitor.java [hunk 2]
```diff
@@ -89,7 +90,9 @@ public ContentDeserializationVisitor(ContentProtos.Content msg) {
 
     private <D extends Described> D visitDescribed(D described) {
         described = visitIdentified(described);
-        described.setPublisher(Sources.fromPossibleKey(msg.getSource()).get());
+
+        Optional<Publisher> possibleSource = Optional.ofNullable(Sources.fromPossibleKey(msg.getSource()).orNull());
+        described.setPublisher(possibleSource.get());  // NPE here is _desired behaviour_
```
Comment (MartinsIrbe) [inline atlas-cassandra/src/main/java/org/atlasapi/content/ContentDeserializationVisitor.java]: Why this is desired behaviour ?

**Comment by MartinsIrbe** [comment]
2 comments otherwise LGTM 👍

File: atlas-api/src/main/java/org/atlasapi/output/writers/RatingsWriter.java (hunk 1 above)
Comment (atish-metabroadcast) [inline atlas-api/src/main/java/org/atlasapi/output/writers/RatingsWriter.java]: Nonnull is done by the IDE. ChecknotNull is done at runtime

File: atlas-cassandra/src/main/java/org/atlasapi/content/ContentDeserializationVisitor.java (hunk 2 above)
Comment (atish-metabroadcast) [inline atlas-cassandra/src/main/java/org/atlasapi/content/ContentDeserializationVisitor.java]: This is the existing legacy publisher behaviour.  I only pull out the publisher to use latter on as part of the review transform.  See the previous line 92

**Pull request merged**